src/System.Private.CoreLib/shared/System/AppDomain.cs

```diff
@@ -47,7 +48,7 @@ public sealed partial class AppDomain : MarshalByRefObject
             remove { AppContext.UnhandledException -= value; }
         }
 
-        public string DynamicDirectory => null;
+        public string? DynamicDirectory => null;
 
         [ObsoleteAttribute("AppDomain.SetDynamicBase has been deprecated. Please investigate the use of AppDomainSetup.DynamicBase instead. https://go.microsoft.com/fwlink/?linkid=14202")]
         public void SetDynamicBase(string path) { }
```

Comment:
Even though it is obsolete, it can accept null right? We still expose it.

Reply:
Yup. Changed.